File: 1-easy/problems/0872-leaf-similar-trees/solution.ts
function leafSimilar(root1: any, root2: any): boolean | number | string | any {
  """
  :type root1: TreeNode
  :type root2: TreeNode
  :rtype: bool
  """
  def dfs(node):
      if (not node) {
          return
      if (not node.left && not node.right) {
          yield node.val
      for (i in dfs(node.left)) {
          yield i
      for (i in dfs(node.right)) {
          yield i
  return all(a == b for a, b in;
             itertools.izip_longest(dfs(root1), dfs(root2)))
}

export { leafSimilar };

File: 1-easy/problems/0872-leaf-similar-trees/solution.py
class Solution:
    def leafSimilar(self, root1, root2):
        """
        :type root1: TreeNode
        :type root2: TreeNode
        :rtype: bool
        """
        def dfs(node):
            if not node:
                return
            if not node.left and not node.right:
                yield node.val
            for i in dfs(node.left):
                yield i
            for i in dfs(node.right):
                yield i
        return all(a == b for a, b in
                   itertools.izip_longest(dfs(root1), dfs(root2)))
